### ISSUE TITLE
When scheduler server extension not installed, error appears in HTML

### ISSUE THREAD

**Issue by whoisltd** [jupyter-server/jupyter-scheduler#329]
## Description

I cant create a schedule in my notebook. When I click the "create" schedule button it shows an error look like this:
![image](https://user-images.githubusercontent.com/41822445/208345594-7e1c8c1f-4c7c-4849-8df0-36c771011529.png)

Error message I get:
```html
<!DOCTYPE HTML> <html> <head> <meta charset="utf-8"> <title>Jupyter Server</title> <link id="favicon" rel="shortcut icon" type="image/x-icon" href="/static/favicon.ico?v=50afa725b5de8b00030139d09b38620224d4e7dba47c07ef0e86d4643f30c9bfe6bb7e1a4a1c561aa32834480909a4b6fe7cd1e17f7159330b6b5914bf45a880"> <link rel="stylesheet" href="/static/style/bootstrap.min.css?v=0e8a7fbd6de23ad6b27ab95802a0a0915af6693af612bc304d83af445529ce5d95842309ca3405d10f538d45c8a3a261b8cff78b4bd512dd9effb4109a71d0ab" /> <link rel="stylesheet" href="/static/style/bootstrap-theme.min.css?v=8b2f045cb5b4d5ad346f6e816aa2566829a4f5f2783ec31d80d46a57de8ac0c3d21fe6e53bcd8e1f38ac17fcd06d12088bc9b43e23b5d1da52d10c6b717b22b3" /> <link rel="stylesheet" href="/static/style/index.css?v=30372e3246a801d662cf9e3f9dd656fa192eebde9054a2282449fe43919de9f0ee9b745d7eb49d3b0a5e56357912cc7d776390eddcab9dac85b77bdb17b4bdae" /> <meta http-equiv="X-UA-Compatible" content="IE=edge" /> <meta name="viewport" content="width=device-width, initial-scale=1.0"> <style type="text/css"> /* disable initial hide */ div#header, div#site { display: block; } </style> </head> <body class="" data-jupyter-api-token="6a23d167d4117a7388d0d7f60cdedc819e905d0353d0f3cc" dir="ltr"> <noscript> <div id='noscript'> Jupyter Server requires JavaScript.<br> Please enable it to proceed. </div> </noscript> <div id="header" role="navigation" aria-label="Top Menu"> <div id="header-container" class="container"> <div id="jupyter_server" class="nav navbar-brand"><a href="/lab?token=6a23d167d4117a7388d0d7f60cdedc819e905d0353d0f3cc" title='dashboard'> <img src='/static/logo/logo.png?v=a2a176ee3cee251ffddf5fa21fe8e43727a9e5f87a06f9c91ad7b776d9e9d3d5e0159c16cc188a3965e00375fb4bc336c16067c688f5040c0c2d4bfdb852a9e4' alt='Jupyter Server' /> </a></div> </div> <div class="header-bar"></div> </div> <div id="site"> <div class="error"> <h1>404 : Not Found</h1> <p>You are requesting a page that does not exist!</p> </div> </div> <script type='text/javascript'> function _remove_token_from_url() { if (window.location.search.length <= 1) { return; } var search_parameters = window.location.search.slice(1).split('&'); for (var i = 0; i < search_parameters.length; i++) { if (search_parameters[i].split('=')[0] === 'token') { // remote token from search parameters search_parameters.splice(i, 1); var new_search = ''; if (search_parameters.length) { new_search = '?' + search_parameters.join('&'); } var new_url = window.location.origin + window.location.pathname + new_search + window.location.hash; window.history.replaceState({}, "", new_url); return; } } } _remove_token_from_url(); </script> </body> </html>

```
## Reproduce

1. Click 'Create Notebook Job'
2. Click 'Create'

## Context

- Operating System and version: ubuntu 22.04
- Browser and version: chrome
- Jupyter Server version: 3.5

<details><summary>Troubleshoot Output</summary>
<pre>
Paste the output from running `jupyter troubleshoot` from the command line here.
You may want to sanitize the paths in the output.

</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
Paste the output from your command line running `jupyter lab` here, use `--debug` if possible.
</pre>
</details>

<details><summary>Browser Output</summary>

<pre>
Paste the output from your browser Javascript console here, if applicable.

</pre>
</details>

**Comment by legasov-broken** [comment]
i have same issue!!!

**Comment by rwuthric** [comment]
I have a similar problem. Only difference: error message is not as detailed. It only shows: "Unexpected error occurred during creation of job." in the red error box instead of the full message as reported by @whoisltd

-   Operating System and version: Ubuntu22.04
-   Browser: Firefox

```
>> jupyter --version
Selected Jupyter core packages...
IPython          : 8.7.0
ipykernel        : 6.19.2
ipywidgets       : 8.0.3
jupyter_client   : 7.4.8
jupyter_core     : 5.1.0
jupyter_server   : 2.0.1
jupyterlab       : 3.5.1
nbclient         : 0.7.2
nbconvert        : 7.2.6
nbformat         : 5.7.0
notebook         : 6.5.2
qtconsole        : not installed
traitlets        : 5.7.1
```

I installed `jupyter_scheduler` with `pip`:
```
pip install jupyter_scheduler
```
and checked that both the `jupyter_scheduler` server extension and the `@jupyterlab/scheduler` prebuilt lab extension are enabled.

**Comment by whoisltd** [comment]
> I have a similar problem. Only difference: error message is not as detailed. It only shows: "Unexpected error occurred during creation of job." in the red error box instead of the full message as reported by @whoisltd
> 
> * Operating System and version: Ubuntu22.04
> * Browser: Firefox
> 
> ```
> >> jupyter --version
> Selected Jupyter core packages...
> IPython          : 8.7.0
> ipykernel        : 6.19.2
> ipywidgets       : 8.0.3
> jupyter_client   : 7.4.8
> jupyter_core     : 5.1.0
> jupyter_server   : 2.0.1
> jupyterlab       : 3.5.1
> nbclient         : 0.7.2
> nbconvert        : 7.2.6
> nbformat         : 5.7.0
> notebook         : 6.5.2
> qtconsole        : not installed
> traitlets        : 5.7.1
> ```
> 
> I installed `jupyter_scheduler` with `pip`:
> 
> ```
> pip install jupyter_scheduler
> ```
> 
> and checked that both the `jupyter_scheduler` server extension and the `@jupyterlab/scheduler` prebuilt lab extension are enabled.

thank you, I fixed this issue as you commented.

**Comment by rwuthric** [comment]
What did you do to solve it? My issue doesn't go away

**Comment by whoisltd** [comment]
> What did you do @whoisltd to solve it? My issue doesn't go away
@rwuthric I use `pip install jupyter_scheduler` to fix my issue. Jupyter scheduler needs both of pip library `jupyter-scheduler`, `@jupyterlab/scheduler`, and **conda** environment on my system to run.

**Comment by guillaumegourlat** [comment]
I have similar problem and I read through the answer given. However I feel it is still a bug. There is a lot of reason why you might not be able to use a conda environment.
If conda environment is a must, shouldn't it had been made explicit in the extension requirement list.
Would it be possible to make the extension work for normal venv ?

Sincerely yours,
Guillaume

**Comment by rwuthric** [comment]
Agree. Feel this issue should not yet be closed yet. I use conda environment to run my JupyterHub and still face the described bug.

**Comment by alexangers** [comment]
Has anyone found a solution to this issue? I still get this error as of 2023-01-21 

![image](https://user-images.githubusercontent.com/123280213/213897076-0725b72d-11e5-45fe-bb11-8e75097c630b.png)

**Comment by susuleung** [comment]
I have the same error in SageMaker Jupyter lab.  I installed using pip and @jupyterlab/scheduler are enable

**Comment by buzzlight** [comment]
I'm having the same problem running the docker image `jupyter/pyspark-notebook`, and getting an interesting difference between `jupyter server extension list` and `jupyter-serverextension list`.  I may be barking up the wrong tree here, but I figure it might be worth mentioning.  

Note the `X is jupyter_scheduler importable?` of the second command

```
(base) jovyan@4fc05219985f:~/work/notebooks$ jupyter server extension list
Config dir: /home/jovyan/.jupyter

Config dir: /opt/conda/etc/jupyter
    jupyter_scheduler enabled
    - Validating jupyter_scheduler...
      jupyter_scheduler 1.2.0 OK
    jupyter_server_terminals enabled
    - Validating jupyter_server_terminals...
      jupyter_server_terminals 0.4.4 OK
    jupyterlab enabled
    - Validating jupyterlab...
      jupyterlab 3.5.3 OK
    nbclassic enabled
    - Validating nbclassic...
      nbclassic 0.4.8 OK
    notebook_shim enabled
    - Validating notebook_shim...
      notebook_shim  OK

Config dir: /usr/local/etc/jupyter

(base) jovyan@4fc05219985f:~/work/notebooks$ jupyter-serverextension list
config dir: /opt/conda/etc/jupyter
    jupyter_scheduler  enabled 
    - Validating...
      X is jupyter_scheduler importable?
    jupyterlab  enabled 
    - Validating...
      jupyterlab 3.5.3 OK
```

**Comment by nilesh0386** [comment]
Go to the Extension Manager and Enable Third Party Extensions. Refer image below
![image](https://user-images.githubusercontent.com/78250828/220856394-c2f4e46c-9d75-4c0f-ad39-16c0f1c6c50b.png)

Restart your JupyterLabs and you should now see the environment properly (earlier only shown as loading...)

Now you can schedule your job

**Comment by rwuthric** [comment]
The extension was enabled since start. Error still present.

**Comment by ellisonbg** [comment]
Hi all, we will dive into this and figure out what is happening. Sorry about the problems.

**Comment by ellisonbg** [comment]
From looking at the error messages, my guess is that this happens because the frontend JupyterLab but the jupyter server extension is not installed.

**Comment by rwuthric** [comment]
In my case this is what returns `jupyter server extension list`
```
Config dir: /opt/jupyterhub/etc/jupyter
    jupyter_scheduler enabled
    - Validating jupyter_scheduler...
Package jupyter_scheduler took 0.1855s to import
      jupyter_scheduler 1.2.0 OK
    jupyter_server_fileid enabled
    - Validating jupyter_server_fileid...
Package jupyter_server_fileid took 0.0024s to import
      jupyter_server_fileid 0.6.0 OK
    jupyter_server_mathjax enabled
    - Validating jupyter_server_mathjax...
Package jupyter_server_mathjax took 0.0010s to import
      jupyter_server_mathjax  OK
    jupyter_server_terminals enabled
    - Validating jupyter_server_terminals...
Package jupyter_server_terminals took 0.0047s to import
      jupyter_server_terminals 0.4.4 OK
    jupyter_server_ydoc enabled
    - Validating jupyter_server_ydoc...
Package jupyter_server_ydoc took 0.0160s to import
      jupyter_server_ydoc 0.6.1 OK
    jupyterlab enabled
    - Validating jupyterlab...
Package jupyterlab took 0.0545s to import
      jupyterlab 3.6.1 OK
    jupyterlab_git enabled
    - Validating jupyterlab_git...
Package jupyterlab_git took 0.0652s to import
      jupyterlab_git 0.41.0 OK
    nbclassic enabled
    - Validating nbclassic...
Package nbclassic took 0.0000s to import
A `_jupyter_server_extension_points` function was not found in nbclassic. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
      nbclassic 0.5.1 OK
    nbdime enabled
    - Validating nbdime...
Package nbdime took 0.0000s to import
      nbdime 3.1.1 OK
    notebook_shim enabled
    - Validating notebook_shim...
Package notebook_shim took 0.0000s to import
A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
      notebook_shim  OK
```
Do I miss anything?

**Comment by rwuthric** [comment]
No problem at all. We all highly appreciate your work. I will do what you recommend. Thank you !
